aptos-move/aptos-transactional-test-harness/src/aptos_test_harness.rs

```diff
@@ -320,37 +314,50 @@ static PRECOMPILED_APTOS_FRAMEWORK_V2: Lazy<PrecompiledFilesModules> = Lazy::new
         .cloned()
         .collect::<Vec<_>>();
 
-    let mut options = move_compiler_v2::Options {
-        sources: prod_sources,
+    let options = move_compiler_v2::Options {
+        sources: prod_sources.clone(),
         dependencies: vec![],
         named_address_mapping: named_address_mapping_strings.clone(),
         known_attributes: aptos_framework::extended_checks::get_all_attribute_names().clone(),
         language_version: None,
         ..move_compiler_v2::Options::default()
     };
 
-    let (_global_env, mut modules) = move_compiler_v2::run_move_compiler_to_stderr(options.clone())
+    let (_global_env, modules) = move_compiler_v2::run_move_compiler_to_stderr(options.clone())
         .expect("framework compilation succeeds");
 
-    options.sources = all_sources;
-    options.language_version = Some(LanguageVersion::latest());
+    PrecompiledFilesModules::new(prod_sources, modules)
+});
+
+static PRECOMPILED_APTOS_FRAMEWORK_V2_WITH_EXPERIMENTAL: Lazy<PrecompiledFilesModules> =
```

Comment:
nit: both lazy initializers are the same moduo language version and experimental filter, can we factor these out into a function that can be called here and for framework only?

Reply:
updated